src/script/script.cpp

```diff
@@ -258,6 +324,44 @@ bool CScript::IsPushOnly() const
     return this->IsPushOnly(begin());
 }
 
+
+bool CScript::IsUnspendable() const
+{
+    if (size() > MAX_SCRIPT_SIZE) {
```

Comment:
did you find anything interesting when you added this check?

Reply:
No as it was previously defined here.

https://github.com/bitcoin/bitcoin/blob/6f03c45f6bb5a6edaa3051968b6a1ca4f84d2ccb/src/script/script.h#L551-L554

The reason I placed it at the top of the if else stack was to avoid doing work on a script that already is invalid due to length. _If the script is too long just mark as unspendable and avoid any extra checking_